### ISSUE TITLE
Option to store sequences from GFF annotation file

### ISSUE THREAD

**Issue by jhumann** [tripal/tripal#1002]
## Feature Request/Discussion
* This feature is Not Urgent

### Description
When we load genomes into chado, we store the GFF and fasta of the assembly and the gff of all the annotated features (gene, mRNA, etc.).  The feature sequences are displayed on the feature pages for genes and mRNA on the website, so chado knows the sequences based on the coordinates from the gff.  But, when the annotations are loaded, the sequence/residues of the features are not parsed and stored.  As a result, I have to extract and load fasta files of genes and mRNA sequences separately in order for those sequences to be downloaded by users on our site.

### Your Specific Use Case
Basically, it would be nice if there was an option when loading a gff of an annotation to have the residues of the features parsed and stored in the chado table at that time, and not require uploading that data separately as a later step.

### Generally Applicable
This will save time when adding genomes to the databases and seems like an easy thing to implement.

**Comment by spficklin** [comment]
So just to add a bit of Chado details to what @jhumann says.  The Tripal GFF loader will only add the sequence for each feature in the GFF file to the `feature.residues` column of Chado if the sequences are in the GFF file at the bottom in a `##FASTA` section.  If they are not, then this requires that the user load the FASTA sequence file of any genes, mRNA, etc through a separate FASTA loading step.

However, if the landmark sequences to which features in the GFF file are aligned (e.g. genes/mRNA/CDS) is present, then we should be able to use existing code to find the sequences (via the `featreloc` table) and add those in automatically when importing the GFF file.  

So, supposing we add a checkbox to the GFF loader that says something like 'Derive missing sequences from alignment with the landmark sequence.'  This would then add all  types of sequences for things like UTRs, CDS, etc... Is this what we want?

**Comment by jhumann** [comment]
A checkbox and possibly a place to limit which features get parsed (e.g. gene, mRNA) would be excellent.  That way dbs can control what gets stored.  Some may want all features, other dbs may only want certain features.

**Comment by laceysanderson** [comment]
I think this is a generic enough enhancement that a contribution to Tripal Core would be welcome 👍 Note: with the switch PMC development focus to Tripal 4 there is no current plans by the PMC to add this functionality.

**Comment by spficklin** [comment]
Hey @jhumann with the Tripal v3.5 release there is a new field that is much smarter about providing sequences.  It will automatically provide the sequence from the reference even if it doesn't have an entry in the `feature.residues` column.  Can you take a look and see if that helps address the problem?  As an aside, the GFF3 loader has been completely rewritten and is lightning fast now :-)  

You will need to enable the new sequence field to show on your mRNA/gene pages.  Previously there were different fields for different types of sequences. This new field takes care of them all in one field.  Try it out and let us know if that helps resolve the problem.

**Comment by laceysanderson** [comment]
Closing this issue as there was no response so we're going to assume the provided fix is sufficient.
